zombie-monitor/src/test/scala/com/broadinstitute/dsp/zombieMonitor/DeletedDiskCheckerSpec.scala

```diff
@@ -0,0 +1,261 @@
+package com.broadinstitute.dsp
+package zombieMonitor
+
+import cats.effect.IO
+import cats.effect.unsafe.implicits.global
+import cats.mtl.Ask
+import com.broadinstitute.dsp.Generators._
+import com.google.api.gax.grpc.GrpcStatusCode
+import com.google.api.gax.rpc.{InternalException, PermissionDeniedException}
+import com.google.cloud.compute.v1
+import io.grpc.Status
+import org.broadinstitute.dsde.workbench.google2.{DiskName, GoogleDiskService, ZoneName}
+import org.broadinstitute.dsde.workbench.model.TraceId
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import org.mockito.ArgumentMatchers.{any, anyLong, anyString}
+import org.mockito.Mockito.{mock, never, verify, when}
+import org.scalatest.flatspec.AnyFlatSpec
+
+class DeletedDiskCheckerSpec extends AnyFlatSpec with CronJobsTestSuite {
+
+  var mockDbReader: DbReader[IO] = _
+  var mockCheckRunnerDeps: CheckRunnerDeps[IO] = _
+  var mockGoogleDiskService: GoogleDiskService[IO] = _
+  var mockDiskCheckerDeps: DiskCheckerDeps[IO] = _
+  var checker: CheckRunner[IO, Disk] = _
+
+  var billingDisabledException = new PermissionDeniedException(
+    new Exception("This API method requires billing to be enabled"),
+    GrpcStatusCode.of(Status.Code.PERMISSION_DENIED),
+    false
+  )
+  var computeEngineNotSetupException = new PermissionDeniedException(
+    new Exception("Compute Engine API has not been used"),
+    GrpcStatusCode.of(Status.Code.PERMISSION_DENIED),
+    false
+  )
+
+  val googleDisk: v1.Disk = com.google.cloud.compute.v1.Disk
+    .newBuilder()
+    .setName("disk-name")
+    .setSizeGb(100)
+    .setType("pd-standard")
+    .build()
+  def setupMocks(): Unit = {
+    mockDbReader = mock(classOf[DbReader[IO]])
+    mockCheckRunnerDeps = mock(classOf[CheckRunnerDeps[IO]])
+    mockGoogleDiskService = mock(classOf[GoogleDiskService[IO]])
+    mockDiskCheckerDeps = DiskCheckerDeps(mockCheckRunnerDeps, mockGoogleDiskService)
+    checker = DeletedDiskChecker.impl(mockDbReader, mockDiskCheckerDeps)
+  }
+
+  it should "call updateDiskStatus with error 'billing is disabled' (isDryRun = false)" in {
+    forAll { (disk: Disk) =>
+      // Arrange
+      setupMocks()
+      when(
+        mockGoogleDiskService.getDisk(any[GoogleProject], ZoneName(anyString()), DiskName(anyString()))(
+          any[Ask[IO, TraceId]]
+        )
+      ).thenAnswer(_ =>
+        IO.raiseError(
+          billingDisabledException
+        )
+      )
+
+      when(mockDbReader.updateDiskStatus(anyLong())).thenReturn(IO.unit)
+
+      // Act
+      val res = checker.checkResource(disk, isDryRun = false)
+      res.unsafeRunSync()
+
+      // Assert
+      verify(mockDbReader).updateDiskStatus(disk.id)
+    }
+  }
+  it should "not call updateDiskStatus with error 'billing is disabled'  (isDryRun = true)" in {
```

Comment:
I'm not sure its necessary to have the `(isDryRun = true)` test for each of the error messages, if `isDryRun` is true it shouldn't update the diskStatus regardless of the error message. Rather than adding a test per exception type, it could be simpler to just have one `not updateDiskStatus if GCP returns an error in a dry run`

Reply:
Since the behavior for the dry run is to never update the disk status, you could have one multi-test with different return types for the mock, something like the following might work:

```  
for (
    getDiskResponse <- List(
                IO.raiseError(billingDisabledException),
                IO.raiseError(computeEngineNotSetupException), 
                IO.pure(None)
)
    
it should s"never update disk status in a dry run" {
...
          .thenAnswer(_ => getDiskResponse)
...
}

```

Reply:
I updated the code to consolidate the test code. I don't think that it is quite as readable, but it is certainly shorter.

Reply:
looks good, thanks for making those changes!